lib/gui/models/drive-constraints.js

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/**
+ * @summary Expose a CLI/GUI shared utility object as an AngularJS service
+ * @module Etcher.Models.DriveConstraints
+ */
+
+const angular = require('angular');
+const MODULE_NAME = 'Etcher.Models.DriveConstraints';
+const DriveConstraintsModel = angular.module(MODULE_NAME, []);
+
+const DriveConstraints = require('../../shared/drive-constraints');
+
+// `DriveConstraintsModel.service` expects a constructor as the second argument, but we want
+// to expose an object. Calling `DriveConstraintsModel.factory` with a a function returning
+// the object is the right way to do it.
+DriveConstraintsModel.factory('DriveConstraintsModel', () => {
```

Comment:
Technically, I could've used `service` here, with a constructor that returns `DriveConstraints` instead of the implicit `this`, but it feels more correct like this, as `AngularJS` won't call this function with `new`.